pkg/build/admission/admission.go

```diff
@@ -107,6 +107,24 @@ func (a *buildByStrategy) checkBuildConfigAuthorization(buildConfig *buildapi.Bu
 	return a.checkAccess(strategyType, subjectAccessReview, attr)
 }
 
+func (a *buildByStrategy) checkBuildRequestAuthorization(req *buildapi.BuildRequest, attr admission.Attributes) error {
+	switch attr.GetResource() {
+	case buildsResource:
+		build, err := a.client.Builds(attr.GetNamespace()).Get(req.Name)
+		if err != nil {
+			return err
+		}
+		return a.checkBuildAuthorization(build, attr)
+	case buildConfigsResource:
+		build, err := a.client.BuildConfigs(attr.GetNamespace()).Get(req.Name)
+		if err != nil {
+			return err
+		}
+		return a.checkBuildConfigAuthorization(build, attr)
+	}
```

Comment:
I would never get that here because if it's not one of the expected resources, we return nil at the beginning:
https://github.com/openshift/origin/pull/4032/files#diff-3eae75d1c69256688c5920b3899663bfL49
Afaik there is no other type of object that we submit for builds other than BuildConfig, Build, and BuildRequest, so having the check at the top is ok. If we want to allow other type of subresource objects in the future, we'd have to explicitly change this code to do that.